packages/headless/src/modules/product/services/index.ts

```diff
@@ -2,46 +2,66 @@ import { HttpStatus, Injectable } from '@nestjs/common';
-        if (isProductExist) {
-
-            return this.helper.serviceResponse.errorResponse('Product already exists', null, HttpStatus.BAD_REQUEST);
-        }
-        else {
-            const newProduct = await this.productRepo.createProduct(product);
-            return this.helper.serviceResponse.successResponse(newProduct);
-        }
+  @validateParams({ schema: ProductCreateSchema })
+  async createProduct(
+    product: Product,
+    extraParam: number,
+  ): Promise<ServiceSuccessResponse | ServiceErrorResponse> {
+    const isProductExist = await this.productRepo.findProduct(product.id);
+    if (isProductExist) {
+      return this.helper.serviceResponse.errorResponse(
+        'Product already exists',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    } else {
+      const newProduct = await this.productRepo.createProduct(product);
+      return this.helper.serviceResponse.successResponse(newProduct);
     }
+  }
 
-    @validateParams({ schema: Joi.string().required() })
-    async getProduct(productId: string): Promise<ServiceSuccessResponse | ServiceErrorResponse> {
-        const foundProduct = await this.productRepo.findProduct(productId);
-        if (foundProduct === null) {
-            return this.helper.serviceResponse.errorResponse('Product not found', { product: ["Not found"] });
-        }
-        else {
-            return this.helper.serviceResponse.successResponse(foundProduct);
-        }
+  @validateParams({ schema: Joi.string().required() })
+  async getProduct(
+    productId: string,
+  ): Promise<ServiceSuccessResponse | ServiceErrorResponse> {
+    const foundProduct = await this.productRepo.findProduct(productId);
+    if (foundProduct === null) {
+      return this.helper.serviceResponse.errorResponse('Product not found', {
+        product: ['Not found'],
+      });
+    } else {
+      return this.helper.serviceResponse.successResponse(foundProduct);
     }
+  }
 
-    @validateParams({ schema: Joi.number().required().label('Skip') }, { schema: Joi.number().required().label('Limit') })
-    async getAllProducts(skip?: number, limit?: number): Promise<ServiceSuccessResponse | ServiceErrorResponse> {
-        const foundProduct = await this.productRepo.findAllProduct(skip, limit);
-        return this.helper.serviceResponse.successResponse(foundProduct);
-    }
+  @validateParams(
+    { schema: Joi.number().required().label('Skip') },
+    { schema: Joi.number().required().label('Limit') },
+  )
+  async getAllProducts(
+    skip?: number,
+    limit?: number,
+  ): Promise<ServiceSuccessResponse | ServiceErrorResponse> {
+    const foundProduct = await this.productRepo.findAllProduct(skip, limit);
+    return this.helper.serviceResponse.successResponse(foundProduct);
+  }
+
+  private test() {
+    ////
+    ///if err
+    return this.helper.serviceResponse.errorResponse('Product not found', {
+      product: ['Not found'],
+    });
+  }
```

Comment:
what's the use case of the function here. please remove unnecessary code if you unwillingly added somewhere